src/harness/harness.ts

```diff
@@ -1031,146 +969,132 @@ namespace Harness {
-
-                const options: ts.CompilerOptions & HarnessOptions = compilerOptions ? ts.clone(compilerOptions) : { noResolve: false };
-                options.target = options.target || ts.ScriptTarget.ES3;
-                options.module = options.module || ts.ModuleKind.None;
-                options.newLine = options.newLine || ts.NewLineKind.CarriageReturnLineFeed;
-                options.noErrorTruncation = true;
-                options.skipDefaultLibCheck = true;
-
-                const newLine = "\r\n";
-                currentDirectory = currentDirectory || Harness.IO.getCurrentDirectory();
-
-                // Parse settings
-                let useCaseSensitiveFileNames = Harness.IO.useCaseSensitiveFileNames();
-                if (harnessSettings) {
-                    setCompilerOptionsFromHarnessSetting(harnessSettings, options);
-                }
-                if (options.useCaseSensitiveFileNames !== undefined) {
-                    useCaseSensitiveFileNames = options.useCaseSensitiveFileNames;
-                }
-                const getCanonicalFileName = ts.createGetCanonicalFileName(useCaseSensitiveFileNames);
-                const inputFilesWithPath = inputFiles.map(f => {
-                    return { unitName: f.unitName, content: f.content, path: ts.toPath(f.unitName, currentDirectory, getCanonicalFileName) };
-                });
-                const otherFilesWithPath = otherFiles.map(f => {
-                    return { unitName: f.unitName, content: f.content, path: ts.toPath(f.unitName, currentDirectory, getCanonicalFileName) };
-                });
+        export function compileFiles(
+            inputFiles: TestFile[],
+            otherFiles: TestFile[],
+            harnessSettings: TestCaseParser.CompilerSettings,
+            compilerOptions: ts.CompilerOptions,
+            // Current directory is needed for rwcRunner to be able to use currentDirectory defined in json file
+            currentDirectory: string): CompilationOutput {
 
-                // Files from built\local that are requested by test "@includeBuiltFiles" to be in the context.
-                // Treat them as library files, so include them in build, but not in baselines.
-                const includeBuiltFiles: TestFileWithPath[] = [];
-                if (options.includeBuiltFile) {
-                    const builtFileName = libFolder + options.includeBuiltFile;
-                    const builtFile: TestFileWithPath = {
-                        unitName: builtFileName,
-                        content: normalizeLineEndings(IO.readFile(builtFileName), newLine),
-                        path: ts.toPath(builtFileName, currentDirectory, getCanonicalFileName)
-                    };
-                    includeBuiltFiles.push(builtFile);
-                }
+            const options: ts.CompilerOptions & HarnessOptions = compilerOptions ? ts.clone(compilerOptions) : { noResolve: false };

```

Comment:
This is sort of weird; is there any reason we don't just define an interface? I guess it's a decent way to use intersection types.